arch/arm/core/aarch64/arm_mmu.c

```diff
@@ -328,6 +328,10 @@ static void init_xlat_tables(const struct arm_mmu_region *region)
 
 /* zephyr execution regions with appropriate attributes */
 static const struct arm_mmu_region mmu_zephyr_regions[] = {
+	MMU_REGION_FLAT_ENTRY("SRAM",
+			      (uintptr_t)CONFIG_SRAM_BASE_ADDRESS,
```

Comment:
@carlocaione is it possible to use here linker symbols?
In the same way the regions, below are defined.
I think this would be consistent, at least with cortex-m

Reply:
E.g. _image_ram_start, or whatever is available for ARM_64....

Reply:
Yes it is possible. I don't have a strong preference here but I honestly thought that using `CONFIG_*` symbols was better than using something exported by the linker.

Reply:
It is not the same. The linker symbols mark the SRAM that is used by the Zephyr image. The CONFIG_ variables map to physical SRAM available in the SoC. These might not match if, say, we're using SRAM partitioning. It is safer to use linker symbols.

That's what is also done right below, with text and RO data areas

Reply:
Using `_image_ram_start` with size `_image_ram_end - _image_ram_start` is not going to work because we need also to map the remaining space between `_image_ram_end` and `_kernel_ram_end` (that in this case is coincident with the end of the SRAM).

So we should map from `_image_ram_start` up to `__kernel_ram_end` as you did on Cortex-M. But that is basically the same as mapping between `CONFIG_SRAM_BASE_ADDRESS` with size `CONFIG_SRAM_SIZE` according to the content of `include/arch/arm/aarch64/scripts/linker.ld` (excluding XIP & co).

For sure I'm missing something here.

Reply:
I am just saying it is more robust using linker symbols because you are always in control of your image. Using Kconfig symbols is fragile because we may change what SRAM_BASE_ADDRESS and SRAM_SIZE mean, actually.

Reply:
I mean, according to this reasoning we should avoid to use `CONFIG_*` symbols all the time in our code because we may change what they mean mid-development? Without considering that there is other code using these symbols including X86 code and MMU code. 

To be clear it's not matter of principle here but since there is no linker script symbol equivalent of `SRAM_SIZE` and I cannot compute it at run/load-time from the linker symbols I have already (because I have a static array of structs for the memory regions) I have to either change how the SRAM region is allocated into the driver or add a new linker symbol myself that I don't think it's much better than using `CONFIG_*`.

Reply:
> we may change what SRAM_BASE_ADDRESS and SRAM_SIZE mean, actually.

I think this is turning into a bikeshed.
There's no equivalent linker symbols for the entire RAM bounds.
We also have a fair amount of code which is already using CONFIG_SRAM_BASE_ADDRESS and CONFIG_SRAM_SIZE.
Where are we are getting the impression that these Kconfigs are going to change?

I think if we want to get rid of these defines in place of something else, file a bug on it.